etcdmain/etcd.go

```diff
@@ -287,9 +287,27 @@ func startEtcdOrProxyV2() {
 	case lerr := <-errc:
 		// fatal out on listener errors
 		if lg != nil {
-			lg.Fatal("listener failed", zap.Error(lerr))
+			lg.Error("listener failed", zap.Error(lerr))
 		} else {
-			plog.Fatal(lerr)
+			plog.Error(lerr)
+		}
+		// It may be the case that several listeners have failed at
+		// once. Exiting as soon as the first one fails would
+		// potentially lose valuable diagnostics, so we wait for a
+		// second and log all available errors on errc before exiting.
+		time.Sleep(1 * time.Second)
+
+		for {
```

Comment:
we should make the outer select in a for loop, and start to wait for a timeout chan. for timeout, we should assign time.After(x) to a initially nil chan when the first error happens.

Reply:
after the timeout chan fires, we can do os.exit(1)